src/opm/parser/eclipse/EclipseState/Schedule/WellConnections.cpp

```diff
@@ -22,24 +22,152 @@
 #include <limits>
 
 #include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/parser/eclipse/EclipseState/Eclipse3DProperties.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/Connection.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/WellConnections.hpp>
 
 namespace Opm {
 
-    WellConnections::WellConnections( std::initializer_list< Connection > cs ) {
-        for( auto&& c : cs ) this->add( c );
+    WellConnections::WellConnections(int headI, int headJ) :
+        headI(headI),
+        headJ(headJ)
+    {
     }
 
 
+
     WellConnections::WellConnections(const WellConnections& src, const EclipseGrid& grid) {
         for (const auto& c : src) {
             if (grid.cellActive(c.getI(), c.getJ(), c.getK()))
                 this->add(c);
         }
     }
 
+    void WellConnections::addConnection(int i, int j , int k ,
+                                        int complnum,
+                                        double depth,
+                                        WellCompletion::StateEnum state ,
+                                        const Value<double>& connectionTransmissibilityFactor,
+                                        const Value<double>& diameter,
+                                        const Value<double>& skinFactor,
+                                        const int satTableId,
+                                        const WellCompletion::DirectionEnum direction)
+    {
+        int conn_i = (i < 0) ? this->headI : i;
+        int conn_j = (j < 0) ? this->headJ : j;
+        Connection conn(conn_i, conn_j, k, complnum, depth, state, connectionTransmissibilityFactor, diameter, skinFactor, satTableId, direction);
+        this->add(conn);
+    }
+
+
+
+    void WellConnections::addConnection(int i, int j , int k ,
+                                        double depth,
+                                        WellCompletion::StateEnum state ,
+                                        const Value<double>& connectionTransmissibilityFactor,
+                                        const Value<double>& diameter,
+                                        const Value<double>& skinFactor,
+                                        const int satTableId,
+                                        const WellCompletion::DirectionEnum direction)
+    {
+        int complnum = -(this->m_connections.size() + 1);
+        this->addConnection(i,
```

Comment:
Can connections be removed from the m_connection list? If so the id could be non-unique

Reply:
No - they can not be removed.